services/src/test/scala/cromwell/services/womtool/DescriberSpec.scala

```diff
@@ -65,5 +71,23 @@ class DescriberSpec extends FlatSpec with Matchers {
     // the end product instead of an intermediate case class hierarchy
     Describer.describeWorkflow(wsfc).asInstanceOf[DescribeSuccess].description.asJson shouldBe expectedJson
   }
+}
 
+object DescriberSpec {
+  sealed trait DescriberSpecTestCase { def expectedDescription: String }
+  final case class FileAndDescription(file: String, override val expectedDescription: String) extends DescriberSpecTestCase
+  final case class UrlAndDescription(url: String, override val expectedDescription: String) extends DescriberSpecTestCase
+
+  def interpretTestCase(caseDirectory: java.io.File): DescriberSpecTestCase = {
+    val description = scala.io.Source.fromFile(caseDirectory.toPath.resolve("description.json").toFile).mkString
+    if (Files.exists(caseDirectory.toPath.resolve("workflow.wdl"))) {
+      val workflow = scala.io.Source.fromFile(caseDirectory.toPath.resolve("workflow.wdl").toFile).mkString
+      FileAndDescription(workflow, description)
+    } else if (Files.exists(caseDirectory.toPath.resolve("workflow_url.txt"))) {
+      val workflowUrl = scala.io.Source.fromFile(caseDirectory.toPath.resolve("workflow_url.txt").toFile).mkString.trim
```

Comment:
why this indirection from the file?

Reply:
Could you rephrase the question?

Reply:
If the question is "why not just list the URLs in the `spec` file" - we have these options:
* The URLs and `description.json`s are in directories
* The URLs and `description.json`s are in the spec file
* The URLs are in the spec file and the expectation `description.json`s are in files in directories.
* Don't store the URL at all - store wdl files in all directories and extrapolate URLs from a known github base path

So:
* I didn't like option 2 because the descriptions are long text files and it's more convenient to keep them as separate json files. 
* I didn't like option 3 because it separates the test case inputs from expectations. If you think of a test as a pair of `(inputs, outputs)`, it feels nice to me to keep that test case in a single place.
* I didn't like option 4 because there's a PR/bootstrapping problem. If I want to add a new test it doesn't exist in `develop` yet, so what do I evaluate to use as my "import by URL" URL.

... but I could probably be talked around if you feel strongly in a different direction